rust/arrow/src/ipc/convert.rs

```diff
@@ -268,22 +270,38 @@ pub(crate) fn get_data_type(field: ipc::Field, may_be_dictionary: bool) -> DataT
             if children.len() != 1 {
                 panic!("expect a list to have one child")
             }
-            DataType::List(Box::new(children.get(0).into()))
+            let child_field = children.get(0);
+            // returning int16 for now, to test, not sure how to get data type
```

Comment:
Perhaps you could change this entire function to something like
```
fn get_data_type_context(field: ipc::Field, may_be_dictionary: bool) -> DataTypeContext {
...
}
```

And then change 
```
pub(crate) fn get_data_type(field: ipc::Field, may_be_dictionary: bool) -> DataType {
    get_data_type_context(field, may_be_dictionary).data_type
```

Reply:
Changed it, but seems like some integration tests are still failing. Any ideas?